server/routes/companies.js

```diff
@@ -2,14 +2,174 @@
 var express = require('express');
 var router = express.Router();
 
+const db = require('../db')
+
 // Returns test string to verify that Companies server is running.
 router.get('/ping', function(req, res, next) {
   res.send('pong - Companies API');
 });
 
-// GET companies listing.
+//Get all companies hiring a specific major
+//written before /: endpoints since those can have various names
+//Need to check query that sends data for both
+router.get('/filter', function(req, res, next) {
+  const pid = req.query.pid;
+  const mid = req.query.mid;
+
+  if(pid == undefined && mid == undefined){
+    res.send('neither!');//not sure how to handle this case
+  }
+  else if(pid == undefined && mid != undefined){
+    db.query('SELECT company_id FROM company_major WHERE major_id = $1;', [mid], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+  } else if(pid != undefined && mid == undefined){
+    db.query('SELECT company_id FROM company_position WHERE position_id = $1;', [pid], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+  } else{ //both
+      db.query('SELECT DISTINCT company_major.company_id FROM company_major LEFT OUTER JOIN company_position ON (position_id = $1 AND major_id = $2);', [pid, mid], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+    }
+});
+
+//Get all companies containing term substring in their name
+router.get('/search', function(req,res,next) {
+  const search = req.query.term;
+  if(search == undefined){
+    res.send('Not Found');
+  }
+
+  db.query('SELECT id FROM company WHERE name ILIKE $1', ['%' + search +'%'], (err, result) => {
```

Comment:
For more concise code, consider `~*`

Reference: https://www.postgresql.org/docs/8.0/functions-matching.html

Reply:
I chose to use ~~* (the equivalent of ILIKE) since the regex expressions used with ~* deviated from the original function of the database call / didn't seem as useful when I tested it. Let me know if we should change this!